au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp

```diff
@@ -0,0 +1,205 @@
+#include "selectionviewcontroller.h"
+
+#include "log.h"
+
+using namespace au::projectscene;
+using namespace au::project;
+using namespace au::processing;
+
+constexpr double MIN_SELECTION_PX = 12.0;
+
+SelectionViewController::SelectionViewController(QObject* parent)
+    : QObject(parent)
+{
+}
+
+void SelectionViewController::onPressed(double x, double y)
+{
+    m_selectionStarted = true;
+    m_startPoint = QPointF(x, y);
+    emit selectionStarted();
+
+    setSelectionActive(false);
+    processingSelectionController()->resetDataSelection();
+}
+
+void SelectionViewController::onPositionChanged(double x, double y)
+{
+    if (!m_selectionStarted) {
+        return;
+    }
+
+    // point
+    QPointF p(x, y);
+    emit selectionChanged(m_startPoint, p);
+
+    // tracks
+    QList<int> tracks = determinateTracks(m_startPoint.y(), y);
+    setSelectedTracks(tracks);
+
+    // time
+    double x1 = m_startPoint.x();
+    double x2 = x;
+    if (x1 > x2) {
+        std::swap(x1, x2);
+    }
+
+    m_context->setSelectionStartTime(m_context->positionToTime(x1));
+    m_context->setSelectionEndTime(m_context->positionToTime(x2));
+}
+
+void SelectionViewController::onReleased(double x, double y)
+{
+    if (!m_selectionStarted) {
+        return;
+    }
+
+    m_selectionStarted = false;
+
+    // point
+
+    QPointF p(x, y);
+
+    emit selectionEnded(m_startPoint, p);
```

Comment:
p is not used anywhere else so maybe it's not worth to create a named variable but rather put it directly into signal argument
emit selectionEnded(m_startPoint,  QPointF(x, y));

Reply:
changed